ngraph/core/src/op/non_max_suppression.cpp

```diff
@@ -530,3 +533,593 @@ void op::v4::NonMaxSuppression::validate_and_infer_types()
+    {
+        const auto num_boxes_boxes = boxes_ps[1];
+        if (num_boxes_boxes.is_static() && scores_ps[0].is_static() && scores_ps[1].is_static())
+        {
+            const auto num_boxes = num_boxes_boxes.get_length();
+            const auto num_classes = scores_ps[1].get_length();
+
+            result[0] = std::min(num_boxes, max_output_boxes_per_class) * num_classes *
+                        scores_ps[0].get_length();
+        }
+    }
+
+    return result;
+}
+
+using V5BoxEncoding = op::v5::NonMaxSuppression::BoxEncodingType;
+
+static void normalize_corner(float* boxes, const Shape& boxes_shape)
+{
+    size_t num_batches = boxes_shape[0];
+    size_t num_boxes = boxes_shape[1];
+
+    float* box_ptr = boxes;
+
+    for (size_t batch = 0; batch < num_batches; ++batch)
+    {
+        for (size_t box = 0; box < num_boxes; ++num_boxes)
+        {
+            float y1 = box_ptr[0];
+            float x1 = box_ptr[1];
+            float y2 = box_ptr[2];
+            float x2 = box_ptr[3];
+
+            float ymin = std::min(y1, y2);
+            float ymax = std::max(y1, y2);
+            float xmin = std::min(x1, x2);
+            float xmax = std::max(x1, x2);
+
+            box_ptr[0] = ymin;
+            box_ptr[1] = xmin;
+            box_ptr[2] = ymax;
+            box_ptr[3] = xmax;
+
+            box_ptr += 4;
+        }
+    }
+}
+
+static void normalize_center(float* boxes, const Shape& boxes_shape)
+{
+    size_t num_batches = boxes_shape[0];
+    size_t num_boxes = boxes_shape[1];
+
+    float* box_ptr = boxes;
+
+    for (size_t batch = 0; batch < num_batches; ++batch)
+    {
+        for (size_t box = 0; box < num_boxes; ++num_boxes)
+        {
+            float x_center = box_ptr[0];
+            float y_center = box_ptr[1];
+            float width = box_ptr[2];
+            float height = box_ptr[3];
+
+            float y1 = y_center - height / 2.0;
+            float x1 = x_center - width / 2.0;
+            float y2 = y_center + height / 2.0;
+            float x2 = x_center + width / 2.0;
+
+            box_ptr[0] = y1;
+            box_ptr[1] = x1;
+            box_ptr[2] = y2;
+            box_ptr[3] = x2;
+
+            box_ptr += 4;
+        }
+    }
+}
+
+static void
+    normalize_box_encoding(float* boxes, const Shape& boxes_shape, const V5BoxEncoding box_encoding)
+{
+    if (box_encoding == V5BoxEncoding::CORNER)
+    {
+        normalize_corner(boxes, boxes_shape);
+    }
+    else
+    {
+        normalize_center(boxes, boxes_shape);
+    }
+}
+
+static std::vector<float> prepare_boxes_data(const HostTensorPtr& boxes,
+                                             const Shape& boxes_shape,
+                                             const V5BoxEncoding box_encoding)
+{
+    element::Type boxes_input_et = boxes->get_element_type();
+
+    size_t boxes_size = shape_size(boxes_shape);
+    std::vector<float> result(boxes_size);
+
+    if (boxes_input_et == ngraph::element::f32)
```

Comment:
What if element type is f64 or any other type?
you support only f32 and f16.

Reply:
Fixed.